include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -50,108 +138,78 @@ namespace fiction
-        /**
-         * X dimension parameter value.
-         */
-        double x;
-        /**
-         * Y dimension parameter value.
-         */
-        double y;
-        /**
-         * Equality operator.
-         *
-         * @param other Other parameter point to compare with.
-         * @return `true` iff the parameter points are equal.
-         */
-        [[nodiscard]] bool operator==(const parameter_point& other) const noexcept
-        {
-            return x == other.x && y == other.y;
-        }
-        /**
-         * Inequality operator.
-         *
-         * @param other Other parameter point to compare with.
-         * @return `true` iff the parameter points are not equal.
-         */
-        [[nodiscard]] bool operator!=(const parameter_point& other) const noexcept
-        {
-            return !(*this == other);
-        }
-        /**
-         * Support for structured bindings.
-         *
-         * @tparam I Index of the parameter value to be returned.
-         * @return The parameter value at the specified index.
-         */
-        template <std::size_t I>
-        auto get() const noexcept
-        {
-            static_assert(I < 2, "Index out of bounds for parameter_point");
-
-            if constexpr (I == 0)
-            {
-                return x;
-            }
-            else  // I == 1
-            {
-                return y;
-            }
-        }
-    };
+    sweep_parameter y_dimension{sweep_parameter::LAMBDA_TF};
     /**
-     * The operational status of the layout for each specified parameter combination. This constitutes the operational
-     * domain. The key of the map is the parameter point, which holds the parameter values in the x and y dimension.
-     * The operational status is stored as the value of the map.
+     * This can store different information depending on the use case. If the operational domain is simulated for
+     * different physical parameters, the parameters are stored with the corresponding operating status.
      */
-    locked_parallel_flat_hash_map<parameter_point, operational_status> operational_values{};
+    locked_parallel_flat_hash_map<Key, Value> operational_domain_values{};
 };
-
+/**
+ * This function compares two parameter points, `p1` and `p2`, for equality with tolerance
+ * (`std::numeric_limits<double>::epsilon()`). It checks if the difference between each coordinate of the parameter
+ * points is within a specified tolerance.
+ *
+ * @param p1 The first parameter point to compare.
+ * @param p2 The second parameter point to compare.
+ * @return `true` if the parameter points are considered equal within the specified tolerance, `false otherwise.
+ */
+static bool check_parameter_points_for_equality(const parameter_point& p1, const parameter_point& p2)
+{
+    constexpr auto tolerance = physical_constants::POP_STABILITY_ERR;
+    return std::abs(p1.x - p2.x) < tolerance && std::abs(p1.y - p2.y) < tolerance;
+}
```

Comment:
Shall we maybe use this implementation for `operator==` in `parameter_point` above? It's silly to compare `double`s directly via `==` anyway.

Reply:
You are right! Good idea.